### PR TITLE
build: update the build system to the latest structure

### PR DESCRIPTION
**Pull request by kiaking**
Close #1253

This PR updates the build system to align with the latest Vue libs. Almost identical to how it is in Vuex 4. The differences are:

- Bundle name is as is.
- `UMD` build instead of `IIFE`.

### CODE REVIEW

**Comment by kiaking** [comment]
Added global `__DEV__` support to prevent issue #1730 from happening.

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,9 @@
...
-TODO.md
-lib
-docs/.vuepress/dist
-examples/**/build.js
-types/typings
-types/test/*.js
-explorations
+/docs/.vuepress/dist
+/examples/**/build.js
+/node_modules
```
Comment (ktsn) [inline .gitignore:3]:
```suggestion
node_modules
```

As `node_modules` should always be ignored.

Comment (kiaking) [reply .gitignore]: Fixed! I think this is same for Vuex 4 so I'll make sure to update that one as well when we merge this.

Comment (ktsn) [reply .gitignore]: Oh thanks. I had overlooked it 😣

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    }
+  }
+
+  if (config.format === 'umd') {
+    c.output.name = c.output.name || 'Vuex'
+  }
+
+  c.plugins.push(replace({
+    __VERSION__: pkg.version,
+    __DEV__: config.format === 'es' && !config.browser
```
Comment (ktsn) [inline rollup.config.js:40]:
```suggestion
    __DEV__: config.format !== 'umd' && !config.browser
```

Commonjs build also need to have `process.env.NODE_ENV` in built file.

File: rollup.main.config.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { createEntries } from './rollup.config'
+
+export default createEntries([
+  { input: 'src/index.js', file: 'dist/vuex.esm.browser.js', format: 'es', browser: true, transpile: false, env: 'development' },
+  { input: 'src/index.js', file: 'dist/vuex.esm.browser.min.js', format: 'es', browser: true, transpile: false, env: 'production' },
```
Comment (ktsn) [inline rollup.main.config.js:5]: Looks like we need `minify: true` here.

File: rollup.config.js (hunk 2 above)
Comment (kiaking) [reply rollup.config.js]: Thanks! Fixed 👍

File: rollup.main.config.js (hunk 3 above)
Comment (kiaking) [reply rollup.main.config.js]: Thanks! Fixed 👍

**Pull request merged**